### DISCUSSION TITLE
Email Provider - failed request, email still being sent

### DISCUSSION THREAD

**Discussion by dpyragas** [nextauthjs/next-auth#7741] [Help]
### Question 💬

Hi all,

I'm using Prisma in my project and added Google auth and email auth. My Verification token model looks like following 

```
model VerificationToken {
  identifier String   @id @default(cuid()) @map("_id")
  token      String   @unique
  expires    DateTime

  @@unique([identifier, token])
}
```

meaning, once user logged in using Google, he wouldn't be able to login using email provider and vice-versa, which I assume is a correct approach?
The issue I'm facing is, that even I'm getting CreateVerificationTokenError and https://next-auth.js.org/errors#signin_email_error once user is trying login via email provider, the email via SendGrid is still being sent.

I have custom signin page and the signin is as following 
```
signIn("email", { email, redirect: false }).then((callback) => {
      if (callback?.ok) {
        toast.success("Please check your email for sign in link");
      } else {
        toast.error("Please use the same login method as first time");
      }
    });
```
    
But this doesn't really work as it doesn't detect the error neither. What would be the correct approach? 
    Thank you for any help.

### How to reproduce ☕️

```
signIn("email", { email, redirect: false }).then((callback) => {
      if (callback?.ok) {
        toast.success("Please check your email for sign in link");
      } else {
        toast.error("Please use the same login method as first time");
      }
    });
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by youminkim** [comment]
I had similar error because of duplicated token value in VerificaitonToken table. The table still had expired tokens. 

Note that I use magic code approach which is few digits codes so high chance of collision
